src/components/AudioPlayer/AudioPlayer.js

```diff
@@ -0,0 +1,205 @@
+  useEffect(() => {
+    if (isPlaying) {
+      audioRef.current.play()
+      startTimer()
+    }
+    if (!isPlaying) {
+      audioRef.current.pause()
+      clearInterval(intervalRef.current)
+    }
+  }, [isPlaying])
+
+  // cleanup when closing or switching audioSrc
+  useEffect(() => {
+    return () => {
+      audioRef.current.pause()
+      clearInterval(intervalRef.current)
+    }
+  }, [audioSrc])
+
+  return (
+    <AudioContainer>
+      <Progressbar
+        type="range"
+        readOnly
+        value={progress}
+        step="1"
+        min="0"
+        max={duration ? duration : `${duration}`} // duration is NaN at first
+      />
+      <TrackInfo>
+        <Image src={image} alt={`track artork for ${title} by ${author}`} />
+        <TitleAndAuthorContainer>
+          <Title>{title}</Title>
+          <Author>{author}</Author>
+        </TitleAndAuthorContainer>
+      </TrackInfo>
+      <AudioControls isPlaying={isPlaying} onPlayPauseClick={setIsPlaying} />
+    </AudioContainer>
+  )
+
+  function startTimer() {
+    // clear any timers
+    clearInterval(intervalRef.current)
+
+    // Pause when audio has ended and if not has ended update the progress bar
+    intervalRef.current = setInterval(() => {
+      if (audioRef.current.ended) audioRef.current.pause()
+      if (!audioRef.current.ended) setProgress(audioRef.current.currentTime)
+      // this function should run every second
+    }, [1000])
+  }
+}
+
+const AudioContainer = styled.div`
+  background: var(--black-accent);
+  width: 100%;
+  display: grid;
+  grid-template-columns: 1fr 2rem;
+  align-items: center;
+  position: relative;
+`
+
+const Progressbar = styled.input`
+  position: absolute;
+  -webkit-appearance: none;
+  background: transparent;
+  height: 0.1rem;
+  top: -0.15rem;
+  left: 0;
+  width: 100%;
+  z-index: 10;
+
+  &::-webkit-slider-thumb {
+    -webkit-appearance: none;
+    opacity: 0;
+  }
+  &::-moz-range-thumb {
+    -webkit-appearance: none;
+    opacity: 0;
+  }
+
+  &::-ms-thumb {
+    -webkit-appearance: none;
+    opacity: 0;
+  }
+
+  &:focus {
+    outline: none;
+  }
+
+  &::-ms-track {
+    background: var(--darkest-grey);
+    border-color: transparent;
+    color: transparent;
+  }
+
+  &::-moz-range-track {
+    background: var(--darkest-grey);
+  }
+
+  &::-moz-range-progress {
+    background: var(--white);
+  }
+  &:focus {
+    outline: none;
+  }
+  &::-webkit-slider-runnable-track {
+    background: var(--darkest-grey);
+    border-color: transparent;
+    color: transparent;
+  }
+  &::-webkit-slider-thumb {
+    opacity: 0;
+    -webkit-appearance: none;
+  }
+  &:focus::-webkit-slider-runnable-track {
+    background: none;
+  }
+
+  &::-ms-fill-lower {
+    background: var(--white);
+  }
+  &::-ms-fill-upper {
+    background: var(--white);
+  }
+
+  &:focus::-ms-fill-lower {
+    background: none;
+  }
+  &:focus::-ms-fill-upper {
+    background: none;
+  }
+`
```

Comment:
didn't go through all of it but since it is a slider it is probably all needed

Reply:
yes unfortunateely this is needed since every browser implements this in y different manner.